src/storage/src/storage_state.rs

```diff
@@ -362,102 +355,6 @@ impl StorageInstanceContext {
     }
 }
 
-/// This maintains an additional read hold on the source data for a sink, alongside
-/// the controller's hold and the handle used to read the shard internally.
-/// This is useful because environmentd's hold might expire, and the handle we use
```

Comment:
FWIW, this is the bit that's no longer true... the critical handle will never expire, and it's responsible for taking the sink's reported frontier into account.